### PR TITLE
Attempting to add install_if

### PR DESCRIPTION
**Pull request by philsturgeon**
Quick stab at getting https://github.com/thoughtbot/appraisal/issues/131 resolved, but im too tired to code up the solution. Here's some tests at least!

Also master didn't have bacon in the gem build list and that was causing my tests to fail!

### CODE REVIEW

**Comment by nickcharlton** [comment]
Hi @philsturgeon!

Nice! Is there anything I can help with this?

**Comment by philsturgeon** [comment]
I honestly have no idea what im doing. 😅

File: lib/appraisal/bundler_dsl.rb [hunk 1]
```diff
@@ -34,6 +34,10 @@ def group(*names, &block)
       @groups[names].run(&block)
     end
 
+    def install_if(*args)
+      raise "Put some code in here"
```
Comment (nickcharlton) [inline lib/appraisal/bundler_dsl.rb]: This is a bit of a fun one, as it's a method call which takes a lambda. I think we should evaluate that and then delegate to the existing `gem` command:

```ruby
def gem(name, *requirements)
  @dependencies.add(name, substitute_git_source(requirements))
end
```

These might help:

* https://github.com/bundler/bundler/blob/6a3b09b2292405cab6b5f4d63004b5c8b782b469/lib/bundler/dsl.rb#L237
* https://github.com/bundler/bundler/blob/3032f4aa447ffa58bfc2fa706472591deb696a66/spec/install/gemfile/install_if.rb#L3

File: spec/acceptance/appraisals_file_bundler_dsl_compatibility_spec.rb [hunk 2]
```diff
@@ -125,6 +133,10 @@
       source "https://other-rubygems.org" do
         gem "sausage"
         gem "pancake"
+
+        install_if -> { true } do
+          gem 'ketchup'
+        end
```
Comment (nickcharlton) [inline spec/acceptance/appraisals_file_bundler_dsl_compatibility_spec.rb]: For testing, I'd add another `describe` (and maybe a context, if we're doing that already) for when it's true and then another when it's false (I've given this less thought than the implementation though).

File: lib/appraisal/bundler_dsl.rb (hunk 1 above)
Comment (philsturgeon) [reply lib/appraisal/bundler_dsl.rb]: Thanks for the feedback! I think evaluating the lambda as this point would be aggressive, as really its about being done when the gemfile is hit. It's used for ENV checks and all sorts of things, as well as ruby checks. 

We dont want the lambda evaluated on generate, we want it evaluated on bundle install. No idea how to do that and im too pneumonia sick to think :D

Comment (nickcharlton) [reply lib/appraisal/bundler_dsl.rb]: 👍 . Were you able to look at this one again? (I hope you're all better now!)

Comment (philsturgeon) [reply lib/appraisal/bundler_dsl.rb]: Back in good health, but I'm still not sure how to do this and tbh I no longer need it. Happy to close, but I think it'd be a good feature for your project.

**Pull request closed without merging**